### ISSUE TITLE
External provisioner misses VolumeSource support for CLONE_VOLUMES 

### ISSUE THREAD

**Issue by humblec** [kubernetes-csi/external-provisioner#172]
CSI 1.0.0 has VolumeSource struct in VolumeContentSource for supporting cloning volumes (https://github.com/container-storage-interface/spec/commit/7a8d2d546db2b010d924708ab36f5dd43a10098d) , but the required changes are not there in the  external-provisioner . 

This is the template we used for testing, am i missing anything here?

> apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: glusterfs-pv-restore
spec:
  storageClassName: glusterfs-csi
  dataSource:
    name: gcs-example-volume
    kind: VolumeSource
    apiGroup: snapshot.storage.k8s.io 
  accessModes:
    - ReadWriteMany
  resources:
    requests:
      storage: 5Gi
>

The error we receive.

~~~
[root@master ~]# kubectl create -f vol-csi.yaml 
The PersistentVolumeClaim "glusterfs-pv-restore" is invalid: spec.dataSource: Unsupported value: "VolumeContentSource.snapshot.storage.k8s.io": supported values: "VolumeSnapshot.snapshot.storage.k8s.io" [root@master ~]# vi vol-csi.yaml [root@master ~]# kubectl create -f vol-csi.yaml The PersistentVolumeClaim "glusterfs-pv-restore" is invalid: spec.dataSource: Unsupported value: "VolumeSource.snapshot.storage.k8s.io": supported values: "VolumeSnapshot.snapshot.storage.k8s.io">
~~~

**Comment by humblec** [comment]
kubernetes used for testing
~~~
[root@master ~]# kubectl version Client Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.0-beta.2", GitCommit:"a4ff09c41589c48547e04f85391aa5610ebe0e17", GitTreeState:"clean", BuildDate:"2018-11-23T00:55:34Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"} Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.0-beta.2", GitCommit:"a4ff09c41589c48547e04f85391aa5610ebe0e17", GitTreeState:"clean", BuildDate:"2018-11-23T00:47:47Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
~~~

**Comment by saad-ali** [comment]
Volume cloning feature for Kubernetes is currently being targetted for an alpha in 1.15. The KEP has been approved: https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20181111-extend-datasource-field.md.

**Comment by msau42** [comment]
Implemented by #220
/close
